src/tests/nested-containers.test.ts

```diff
@@ -0,0 +1,650 @@
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for optional< pair< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setop('alice', {'key': 60, 'value': 61}) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('bob').setop('bob', null) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for optional< tuple< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setot('alice', [1001,2001]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('bob').setot('bob', null) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for map< set< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setmst('alice', [{'key': 1,'value': [10, 10, 12, 16]},  {'key': 2, 'value': [200, 300]} ]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for map< vector< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setmv('alice', [{'key': 1, 'value': [10, 10, 12, 16]},  {'key': 2, 'value': [200, 300]} ]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for map< optional< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setmo('alice', [{'key': 10, 'value': 1000}, {'key': 11, 'value': null}]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for map< map< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setmm('alice', [{'key': 10, 'value': [{'key': 200, 'value': 2000}, {'key': 201, 'value': 2001}] }, {'key': 11, 'value': [{'key': 300, 'value': 3000}, {'key': 301, 'value': 3001}] } ]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for map< pair< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setmp('alice', [{'key': 36, 'value': {'key': 300, 'value': 301}}, {'key': 37, 'value': {'key': 600, 'value': 601}} ]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for map< tuple< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setmt('alice', [{'key':1,'value':[10,11]},  {'key':2,'value':[200,300]} ]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for pair< set< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setpst('alice', {'key': 20, 'value': [200, 200, 202]}) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
```

Comment:
Follow the same rule to use first-second only for setpst,setpv, setpo, setpm,setpp,setpt